### PR TITLE
MRG: Add "highlight" parameter to Evoked.plot() to conveniently highlight time periods

### PR DESCRIPTION
**Pull request by hoechenberger**
Allows to conveniently produce figures like:

evoked

Becomes really powerful when combined with `Evoked.plot_topomap()`'s `average` parameter:

joint

MWE:
```python
# %%
from pathlib import Path
import mne


sample_dir = Path(mne.datasets.sample.data_path())
sample_fname = sample_dir / 'MEG' / 'sample' / 'sample_audvis_raw.fif'

raw = mne.io.read_raw_fif(sample_fname)
raw.crop(tmax=60)

events = mne.find_events(raw, stim_channel='STI 014')
event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
            'visual/right': 4, 'face': 5, 'buttonpress': 32}

epochs = mne.Epochs(raw, events=events, event_id=event_id,
                    tmin=-0.2, tmax=0.5, baseline=(None, 0),
                    preload=True)

evoked = epochs.average()

evoked.plot(
    spatial_colors=True, picks='mag', gfp=True,
    highlight=[(0.05, 0.13), (0.23, 0.3)]
);
evoked.plot_joint(
    times=[0.09],
    topomap_args=dict(average=0.1),
    ts_args=dict(highlight=(0.04, 0.14)),
    picks='mag'
);
```

### CODE REVIEW

File: mne/viz/evoked.py [hunk 1]
```diff
@@ -749,6 +772,18 @@ def plot_evoked(evoked, picks=None, exclude='bads', unit=True, show=True,
 
         .. versionadded:: 0.16
     %(sphere_topomap_auto)s
+    highlight : array-like of float, shape(2,) | array-like of tuples of float, shape (n, 2) | None
```
Comment (hoechenberger) [inline mne/viz/evoked.py:775]:
```suggestion
    highlight : array-like of float, shape(2,) | array-like of float, shape (n, 2) | None
```

**Comment by agramfort** [comment]
any idea why it's not taking the full y range?

**Comment by hoechenberger** [comment]
> any idea why it's not taking the full y range?

Nope, it might be that we're doing some axis-range foo somewhere which I've overlooked. Any help on this would be appreciated! But I think even like it is now it's a step forward :)

**Comment by hoechenberger** [comment]
Rendered tutorial:

https://output.circle-artifacts.com/output/job/18c37ac7-3c9c-4be8-9867-03e9c643208b/artifacts/0/dev/auto_tutorials/evoked/20_visualize_evoked.html#plotting-signal-traces

**Comment by agramfort** [comment]
fix pushed

**Comment by hoechenberger** [comment]
Amazing!!!!!

**Comment by hoechenberger** [comment]
Rendered tutorial:
https://output.circle-artifacts.com/output/job/4f2123f0-82d0-4f9e-a419-15b76f5e6d85/artifacts/0/dev/auto_tutorials/evoked/20_visualize_evoked.html#plotting-signal-traces

**Comment by hoechenberger** [comment]
@agramfort Please merge if happy

**Review by agramfort — APPROVED** [review]
anyone merge if happy.

thx @hoechenberger

**Pull request merged**